src/frontend/js/contexts/SessionContext/JoanieSessionProvider.tsx

```diff
@@ -47,6 +47,20 @@ const JoanieSessionProvider = ({ children }: React.PropsWithChildren<{}>) => {
   });
   const previousUserState = usePrevious(user);
 
+  /**
+   * It is on purpose that this side effect is done synchronously and not in a useEffect.
+   * If we had wrapped it inside a useEffect it would cause race conditions where useSession
+   * queries would turn enabled == true before the side effect is done and then try to use
+   * the sessionStorage RICHIE_USER_TOKEN before it is set in fetchWithJWT. Thus causing 401 errors.
+   */
+  if (previousUserState !== user) {
+    if (user) {
+      sessionStorage.setItem(RICHIE_USER_TOKEN, user?.access_token!);
+    } else {
+      sessionStorage.removeItem(RICHIE_USER_TOKEN);
+    }
+  }
+
```

Comment:
Indeed, it looks weird 😬 

Maybe we could put into the request method ?

If the response is successful, we overwrite the token. If the response is 401, we remove the token? WDYT ?